Godeps/_workspace/src/k8s.io/kubernetes/pkg/client/transport/round_trippers.go

```diff
@@ -130,6 +133,36 @@ func (rt *basicAuthRoundTripper) CancelRequest(req *http.Request) {
 
 func (rt *basicAuthRoundTripper) WrappedRoundTripper() http.RoundTripper { return rt.rt }
 
+type actAsRoundTripper struct {
+	actAs string
+	rt    http.RoundTripper
+}
+
+// NewActAsRoundTripper will apply a BASIC auth authorization header to a
+// request unless it has already been set.
+func NewActAsRoundTripper(actAs string, rt http.RoundTripper) http.RoundTripper {
+	return &actAsRoundTripper{actAs, rt}
+}
+
+func (rt *actAsRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	if len(req.Header.Get("Act-As")) != 0 {
+		return rt.rt.RoundTrip(req)
+	}
+	req = cloneRequest(req)
+	req.Header.Set("Acting-As", rt.actAs)
```

Comment:
I think my preference is for "Kubernetes-Acting-As".

On Mon, Mar 28, 2016 at 10:50 AM, Eric Tune <notifications@github.com>
wrote:

> In
> Godeps/_workspace/src/k8s.io/kubernetes/pkg/client/transport/round_trippers.go
> <https://github.com/openshift/origin/pull/8006#discussion_r57604233>:
>
> > +	actAs string
> > +	rt    http.RoundTripper
> > +}
> > +
> > +// NewActAsRoundTripper will apply a BASIC auth authorization header to a
> > +// request unless it has already been set.
> > +func NewActAsRoundTripper(actAs string, rt http.RoundTripper) http.RoundTripper {
> > +	return &actAsRoundTripper{actAs, rt}
> > +}
> > +
> > +func (rt *actAsRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
> > +	if len(req.Header.Get("Act-As")) != 0 {
> > +		return rt.rt.RoundTrip(req)
> > +	}
> > +	req = cloneRequest(req)
> > +	req.Header.Set("Acting-As", rt.actAs)
>
> @lavalamp <https://github.com/lavalamp> @bgrant0607
> <https://github.com/bgrant0607> do you have a preference on what prefix
> to use for kubernetes-api-specific headers?
>
> @deads2k <https://github.com/deads2k> @smarterclayton
> <https://github.com/smarterclayton> do you mind an X-Kubernetes-* or
> X-Kube-* or X-K8s-* prefix for http headers that are common to kubernetes
> and Openshift?
>
> The reason this is a header is that (1) it is for authentication, which
> already uses headers, and (2), it applies to almost all requests, so adding
> a query parameter to all resources and methods would be pretty verbose.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8006/files/593c28e562a84707f05dd866298977badb7d7b6f#r57604233>
>